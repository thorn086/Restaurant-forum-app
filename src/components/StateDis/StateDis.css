


.state_list{
    
    width: 95%;
    margin: .5rem;
    height: auto;
   
}
.state-list-items{
    list-style-type: none;
    padding: 0;
    margin: 0;
}
.state-items{
    border: 1px solid black;
    background-color: rgba(70, 70, 193, 0.9);
    margin:2rem 3rem;
    border-radius: 5%;
    box-shadow: 3px 3px rgb(61, 61, 95);
    padding: 1rem 1rem 1.5rem 1rem;
    text-align: center;
    height: auto;
    letter-spacing: 2px;
}
.back-btn{
    border: 1px solid black;
    background-color: rgba(98, 98, 250, 0.9);
    text-shadow: -1px -1px black;
    margin:1rem 3rem;
    border-radius: 8%;
    box-shadow: 3px 3px rgb(61, 61, 95);
    padding: 1rem 1rem 1.5rem 1rem;
    text-align: center;
    height: 3rem;
    letter-spacing: 2px;
    width: 12.8rem;
    border-radius: 6%;
    color: white;
    font-size: 16px;
    font-family: 'Fjalla One', sans-serif;
    font-weight: 500;
    text-transform: uppercase;

}
.state-list-link{
    text-decoration: none;
    color: white;
}