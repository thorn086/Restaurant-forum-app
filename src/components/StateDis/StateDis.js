import React from 'react'
import { withRouter, NavLink } from 'react-router-dom'
import RestForumContext from '../../context'
import './StateDis.css'

class StateDis extends React.Component {
    static contextType = RestForumContext
    render() {
        const { states } = this.context;
        return (
            <div>
                <div className='state_list'>
                    <ul className='state-list-items'>
                        {states.map(newState =>
                            <li key={newState.id} className='state-items'>
                                <NavLink className='state-list-link'
                                    to={`/states/${newState.id}`}>
                                    <h2 className='state-name'>
                                        {newState.name}
                                    </h2>
                                </NavLink>
                            </li>
                        )}
                    </ul>
                    <button className='state-btn back-btn ' tag='button' onClick={() => this.props.history.goBack()}>Back</button>

                </div>

            </div>
        )
    }
}
export default withRouter(StateDis)