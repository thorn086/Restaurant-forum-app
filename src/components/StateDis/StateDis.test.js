import React from 'react';
import StateDis from './StateDis';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('StateDis', () => {
    const stateDis = shallow(<StateDis/>);
  
    it('renders properly', () => {
        expect(stateDis).toMatchSnapshot();
    });
});