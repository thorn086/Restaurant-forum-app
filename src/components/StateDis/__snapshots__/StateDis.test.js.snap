// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StateDis renders properly 1`] = `ShallowWrapper {}`;
